### PR TITLE
fix(frontend): Remove redundant join non-eq pullup

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Remove redundant join non-eq pullup

## Checklist For Contributors

- [X] I have written necessary rustdoc comments
- [X] I have added necessary unit tests and integration tests
- [X] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [X] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [X] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/logical_join.rs [hunk 1]
```diff
@@ -1189,33 +1158,7 @@ impl LogicalJoin {
         logical_join: LogicalJoin,
     ) -> Result<PlanRef> {
         assert!(predicate.has_eq());
-        // Convert to Hash Join for equal joins
```
Comment (jon-chuang) [inline src/frontend/src/optimizer/plan_node/logical_join.rs:1192]: Perhaps it's ok to remove for stream hash join but not batch hash join

Comment (jon-chuang) [reply src/frontend/src/optimizer/plan_node/logical_join.rs:1192]: Turns out it's just a test.

**Review by st1page — COMMENTED** [review]
I think the pulling up is a kind of optimization because we think the Filter operator can process more efficiently.

**Comment by jon-chuang** [comment]
> I think the pulling up is a kind of optimization because we think the Filter operator can process more efficiently.

Oh, I see, as it is vectorized, correct?

**Comment by jon-chuang** [comment]
I will add some comments instead so it is clearer.

**Comment by lmatz** [comment]
> I think the pulling up is a kind of optimization because we think the Filter operator can process more efficiently.

Possible, Flink has a different strategy than us: https://github.com/risingwavelabs/risingwave/issues/7343#issuecomment-1381456708. Flink has no vectorized execution.

**Comment by jon-chuang** [comment]
Actually, I notice that the vectorized part is already implemented within the executor for BatchHashJoin.
https://github.com/risingwavelabs/risingwave/blob/7cadc39dffda32a01587a08913591cf9666fe84e/src/batch/src/executor/join/hash_join.rs#L362

Turns out we have not been using this code path :P So let's apply this change for batch hash join after all.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Comment by lmatz** [comment]
Alright, `license-header-check` stops working again......

Maybe let @TennyZhuang force merge it after the conflicts are resolved.

**Pull request merged**